tests/Integration/Behaviour/Features/Context/CartRuleFeatureContext.php

```diff
@@ -140,6 +140,7 @@ protected function createCartRule(
         $now->add(new DateInterval('P1Y'));
         $cartRule->date_to = $now->format('Y-m-d H:i:s');
         $cartRule->active = 1;
+        $cartRule->code = '';
```

Comment:
After this change, any Behat tests that uses `createCartRule()` will create Cart Rules that have no code and, as you mentioned, cart rules with no codes are automatically applied when eligible.

I'm a bit afraid of the consequences of this change. 🤷‍♂️ However CI seems green ... so maybe there's no issue.